osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -187,7 +229,27 @@ void genDetailsFromAddr(const struct ifaddrs* addr, QueryData& results) {
     r["odrops"] = INTEGER(0);
     r["collisions"] = BIGINT_FROM_UINT32(ifd->ifi_collisions);
     r["last_change"] = BIGINT_FROM_UINT32(ifd->ifi_lastchange.tv_sec);
-#endif
+    if (context.isColumnUsed("link_speed")) {
+      int fd = socket(AF_INET, SOCK_DGRAM, 0);
+      if (fd >= 0) {
+        struct ifmediareq ifmr;
```

Comment:
For other reviewers, here is the struct definition from Apple's master branch of XNU: https://github.com/apple/darwin-xnu/blob/master/bsd/net/if.h#L576